flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultSubpartition.java

```diff
@@ -115,52 +115,62 @@ protected Throwable getFailureCause() {
 
 	public abstract boolean isReleased();
 
+	public abstract boolean isFinished();
+
+	public abstract boolean isFlushRequested();
```

Comment:
The same question as above. Can it be non public or mark it unsafe?

Reply:
Could be `protected` as above.